Create 2.43.2.md (#1320)
percona/pmm-doc: docs/release-notes/2.43.2.md

# Percona Monitoring and Management 2.43.2

| **Release date** | October 22nd, 2024                                                                                     |
| -----------------| ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/index.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them, and execute database management operations regardless of whether your databases are located on-premises or in the cloud.

##  PostgreSQL 17 support for Query Analytics (QAN)

This patch release adds PostgreSQL 17 support for Query Analytics with updates to the `pg_stat_statements` and `pg_stat_monitor` extensions.

> To align with PostgreSQL 17 standards, we've renamed the `blk_read_time` field to `shared_blk_read_time` and the `blk_write_time` field to `shared_blk_write_time`.

Reviewer: We also added `local_blk_read_time` and `local_blk_write_time` metrics, however we don't use them on dashboards yet, so probably we can omit them in RN.

Author: Thanks, Artem. I've confirmed with @BupycHuk  that we don't want to mention it yet.

# Percona Monitoring and Management 2.43.2

| **Release date** | October 30th, 2024                                                                                     |
| -----------------| ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/index.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them, and execute database management operations regardless of whether your databases are located on-premises or in the cloud.

## PostgreSQL 17 support for Query Analytics (QAN)

This patch release adds PostgreSQL 17 support for Query Analytics with updates to the `pg_stat_statements` and `pg_stat_monitor` extensions.

To align with PostgreSQL 17 standards, we've renamed the `blk_read_time` field to `shared_blk_read_time` and the `blk_write_time` field to `shared_blk_write_time`.

> All dashboards reflect these changes, and the API supports both old and new field names, for backwords compatibility.

Reviewer 2: ```suggestion
All dashboards reflect these changes, and the API supports both old and new field names, for backward compatibility.
```

## Secure Grafana image rendering

PMM Server now supports secure Grafana image rendering capabilities through a dedicated container deployment, providing isolated rendering operations without impacting PMM Server resources.

Previously installed directly within PMM Server, the Grafana Image Renderer plugin now runs in a separate container, offering secure HTTPS communication and custom CA certificate configuration through the PMM API.

To update the plugin installation:

1. Deploy the Grafana Image Renderer container (`grafana/grafana-image-renderer:latest`) alongside PMM Server within the same Docker network.
2. Configure PMM Server with the following environment variables, where `renderer` is the hostname of the Grafana Image Renderer container, and `pmm-server` is the hostname of PMM Server within the Docker network: 

    - `GF_RENDERING_SERVER_URL=http://renderer:8081/render`
    - `GF_RENDERING_CALLBACK_URL=https://pmm-server:8443/graph/`

For more information, see [Share dashboards and panels](../how-to/share-dashboard%202.md).

## Fixed: High memory consumption in MongoDB exporter

> We have also updated the MongoDB exporter to address a critical issue present in version 2.43.1 where unclosed connections led to increasingly high memory consumption over time.

Reviewer 2: ```suggestion
We have updated the MongoDB exporter to address a critical issue present in version 2.43.1 where unclosed connections led to increasingly high memory consumption over time.
```

> This resolves the *`Failed to get PBM configuration`* error that contributed to memory issues, and significantly reduces the rate of increase in total memory allocations.

Reviewer 2: ```suggestion
This resolves the *`Failed to get PBM configuration`* error, the users may have encountered as a result of the memory leak, and significantly reduces the rate of increase in total memory allocations.
```
